pkg/api/vlabs/types.go

```diff
@@ -408,24 +408,25 @@ type Extension struct {
-	Ports                  []int                `json:"ports,omitempty" validate:"dive,min=1,max=65535"`
-	AvailabilityProfile    string               `json:"availabilityProfile"`
-	ScaleSetPriority       string               `json:"scaleSetPriority,omitempty" validate:"eq=Regular|eq=Low|len=0"`
-	ScaleSetEvictionPolicy string               `json:"scaleSetEvictionPolicy,omitempty" validate:"eq=Delete|eq=Deallocate|len=0"`
-	StorageProfile         string               `json:"storageProfile" validate:"eq=StorageAccount|eq=ManagedDisks|len=0"`
-	DiskSizesGB            []int                `json:"diskSizesGB,omitempty" validate:"max=4,dive,min=1,max=1023"`
-	VnetSubnetID           string               `json:"vnetSubnetID,omitempty"`
-	IPAddressCount         int                  `json:"ipAddressCount,omitempty" validate:"min=0,max=256"`
-	Distro                 Distro               `json:"distro,omitempty"`
-	KubernetesConfig       *KubernetesConfig    `json:"kubernetesConfig,omitempty"`
-	ImageRef               *ImageReference      `json:"imageReference,omitempty"`
-	Role                   AgentPoolProfileRole `json:"role,omitempty"`
+	Name                      string               `json:"name" validate:"required"`
+	Count                     int                  `json:"count" validate:"required,min=1,max=100"`
+	VMSize                    string               `json:"vmSize" validate:"required"`
+	OSDiskSizeGB              int                  `json:"osDiskSizeGB,omitempty" validate:"min=0,max=1023"`
+	DNSPrefix                 string               `json:"dnsPrefix,omitempty"`
+	OSType                    OSType               `json:"osType,omitempty"`
+	Ports                     []int                `json:"ports,omitempty" validate:"dive,min=1,max=65535"`
+	AvailabilityProfile       string               `json:"availabilityProfile"`
+	ScaleSetPriority          string               `json:"scaleSetPriority,omitempty" validate:"eq=Regular|eq=Low|len=0"`
+	ScaleSetEvictionPolicy    string               `json:"scaleSetEvictionPolicy,omitempty" validate:"eq=Delete|eq=Deallocate|len=0"`
+	StorageProfile            string               `json:"storageProfile" validate:"eq=StorageAccount|eq=ManagedDisks|len=0"`
+	DiskSizesGB               []int                `json:"diskSizesGB,omitempty" validate:"max=4,dive,min=1,max=1023"`
+	VnetSubnetID              string               `json:"vnetSubnetID,omitempty"`
+	IPAddressCount            int                  `json:"ipAddressCount,omitempty" validate:"min=0,max=256"`
+	Distro                    Distro               `json:"distro,omitempty"`
+	KubernetesConfig          *KubernetesConfig    `json:"kubernetesConfig,omitempty"`
+	ImageRef                  *ImageReference      `json:"imageReference,omitempty"`
+	Role                      AgentPoolProfileRole `json:"role,omitempty"`
+	AcceleratedNetworkEnabled bool                 `json:"acceleratedNetworkEnabled,omitempty"`
```

Comment:
Could you please add this new property to `clusterdefinition.md`?

Reply:
Done.